### PR TITLE
Serialize scalariform intents

### LINKED ISSUE #197: Scalariform custom setting breaks ensimeConfig
Using a non-boolean (or Int) setting in Scalariform throws a MatchError from `ensimeConfig`.

## Steps to reproduce
1. Add following Scalariform settings to a project's settings

    ```scala
    ScalariformKeys.preferences := ScalariformKeys.preferences.value
      .setPreference(DanglingCloseParenthesis, Preserve)
                                               ^^^
                                               notice the non bool or int value
    ```

2. Run `sbt ensimeConfig`

3. Watch it die with a `MatchError`

## Underlying issue
The part that generates the contents of the `.ensime` configuration file, `SExpFormatter` has a non-exhaustive match, [as highlighted here, on lines 745-749](https://github.com/ensime/ensime-sbt/blob/068b5efb47878802abb486cdb4930564d708f194/src/main/scala/EnsimePlugin.scala#L745-L749)

The fix is probably pretty simple, however since the `.ensime` configuration file is also read from the server, I wanted to post an issue before submitting a PR.

### PR DESCRIPTION
**Pull request by jodersky**
Workaround to serialize scalariform intents. Fixes #197
Note that this is a two-part PR: ensime/ensime-server#1448 is also required on the server side

### CODE REVIEW

**Comment by fommil** [comment]
Oh interesting, you've considered and worked around the version problem. If this goes green I'm happy to merge, but can you please add a test?

**Comment by jodersky** [comment]
ok will do

**Comment by jodersky** [comment]
added a test

File: src/main/scala/EnsimePlugin.scala [hunk 1]
```diff
@@ -743,10 +743,16 @@ object SExpFormatter {
...
-      case (desc, v: Boolean) =>
-        s":${desc.key} ${toSExp(v)}"
-      case (desc, v: Int) =>
-        s":${desc.key} $v"
+      case (desc, value) =>
+        val vs = value match {
+          case b: Boolean => toSExp(b)
+          case i: Int     => i.toString
+          case intent =>
+            // quick fix to serialize intents, until the scalariform dependency is
```
Comment (fommil) [inline src/main/scala/EnsimePlugin.scala]: I'll forgive your Simplified English spelling

File: src/sbt-test/ensime-sbt/scalariform/build.sbt [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+SbtScalariform.scalariformSettings
+
+ScalariformKeys.preferences := FormattingPreferences
+  .setPreference(DoubleIndentClassDeclaration, true)
+  .setPreference(DanglingCloseParenthesis, Preserve)
+  .setPreference(MultilineScaladocCommentsStartOnFirstLine, false)
+  .setPreference(PlaceScaladocAsterisksBeneathSecondAsterisk, true)
+  .setPreference(AlignSingleLineCaseStatements.MaxArrowIndent, 20)
+
+scalaVersion := "2.11.7"
```
Comment (fommil) [inline src/sbt-test/ensime-sbt/scalariform/build.sbt]: don't change the scalaVersion, use the default so you hit the drone cache.

File: src/sbt-test/ensime-sbt/scalariform/test [hunk 3]
```diff
@@ -0,0 +1,11 @@
+> ensimeConfig
+$ absent scalaVersionAtStartupWarning
+> ensimeExpect .ensime ensimeConfig.expect
+> ensimeConfigProject
+$ absent scalaVersionAtStartupWarning
+> ensimeExpect project/.ensime ensimeConfigProject.expect
+> checkJavaOptions
+> debugging
+> checkJavaOptions -agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=5005
```
Comment (fommil) [inline src/sbt-test/ensime-sbt/scalariform/test]: you don't need to check the debugging settings, or the scala version warning.

File: src/sbt-test/ensime-sbt/scalariform/build.sbt (hunk 2 above)
Comment (jodersky) [inline src/sbt-test/ensime-sbt/scalariform/build.sbt]: I just copied this from the simple test
:sob: I'll have to update all the other fields in .ensime then

File: src/main/scala/EnsimePlugin.scala (hunk 1 above)
Comment (jodersky) [inline src/main/scala/EnsimePlugin.scala]: do you mean the "zee" in serialise :stuck_out_tongue_closed_eyes:  ?

File: src/sbt-test/ensime-sbt/scalariform/build.sbt (hunk 2 above)
Comment (fommil) [inline src/sbt-test/ensime-sbt/scalariform/build.sbt]: oh darn, I didn't realise we did that. ok, leave it then, or update to 2.11.8 if you're really brave.

**Comment by jodersky** [comment]
I removed the debugging options (took me 3 tries, sorry if I flooded CI please cancel all except the last build).
Regarding the scala version, all other tests I check actually use 2.11.7

**Comment by fommil** [comment]
weird, drone is actually green

**Comment by fommil** [comment]
thanks! Nice hack

**Pull request merged**